src/pages/Post.jsx

```diff
@@ -1,41 +1,359 @@
-import { useEffect, useState } from "react";
+import { useNavigate } from "@tanstack/react-router";
+
+import { useState, useEffect } from "react";
+
+import Navigation from "../components/navbar";
+
+ 
 
 const initialPostState = {
```

Comment:
Not sure where you learnt this pattern, I just hope you understand why it's preferred.